js/private/js_image_layer.mjs.tpl

```diff
@@ -0,0 +1,353 @@
+import { readdir, readFile, readlink, writeFile } from 'node:fs/promises'
```

Comment:
This being a `.tpl` is really annoying, and breaks the diff (or maybe there is no usable diff, I can't tell atm).

Which part is a template? Can we do some other type of magic to keep the `.mjs` extension despite using it as a template?

Reply:
The formatting is also odd, with tabs/spaces mixed all over the place etc